packages/foundry/src/vault/adapter/convex/ConvexAdapter.sol

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: GPL-3.0
+// Docgen-SOLC: 0.8.15
+
+pragma solidity ^0.8.15;
+
+import { AdapterBase, IERC20, IERC20Metadata, SafeERC20, ERC20, Math, IStrategy, IAdapter } from "../abstracts/AdapterBase.sol";
+import { WithRewards, IWithRewards } from "../abstracts/WithRewards.sol";
+import { IConvexBooster, IBaseRewarder } from "./IConvex.sol";
+
+/**
+ * @title   Convex Adapter
+ * @author  amatureApe
+ * @notice  ERC4626 wrapper for Convex Vaults.
+ *
+ * An ERC4626 compliant Wrapper for https://github.com/beefyfinance/beefy-contracts/blob/master/contracts/BIFI/vaults/BeefyVaultV6.sol.
+ * Allows wrapping Convex Vaults with or without an active Booster.
+ * Allows for additional strategies to use rewardsToken in case of an active Booster.
+ */
+contract ConvexAdapter is AdapterBase, WithRewards {
+  using SafeERC20 for IERC20;
+  using Math for uint256;
+
+  string internal _name;
+  string internal _symbol;
+
+  /// @notice The poolId inside Convex booster for relevant Curve lpToken.
+  uint256 public pid;
+
+  /// @notice The booster address for Convex
+  IConvexBooster public booster;
+
+  /// @notice The Convex BaseRewarder.
+  IBaseRewarder public baseRewarder;
+
+  /*//////////////////////////////////////////////////////////////
+                            INITIALIZATION
+    //////////////////////////////////////////////////////////////*/
+
+  /**
+   * @notice Initialize a new Convex Adapter.
+   * @param adapterInitData Encoded data for the base adapter initialization.
+   * @param convexInitData Encoded data for the convex adapter initialization.
+   * @dev `_booster` - The booster address for Convex.
+   * @dev `_pid` - The poolId for lpToken.
+   * @dev This function is called by the factory contract when deploying a new vault.
+   */
+  function initialize(
+    bytes memory adapterInitData,
+    address,
```

Comment:
We can use this contract https://etherscan.io/address/0xF403C135812408BFbE8713b5A23a04b3D48AAE31 as the registry to confirm a correct pid and asset etc.